src/multi-column-adapter/src/test/scala/MultiColumnAdapterSpec.scala

```diff
@@ -3,47 +3,65 @@
 
 package com.microsoft.ml.spark
 
-import org.apache.spark.ml.feature.Tokenizer
+import org.apache.spark.ml.feature.{StringIndexer, Tokenizer}
 import com.microsoft.ml.spark.schema.DatasetExtensions._
-import org.apache.spark.ml.Transformer
+import org.apache.spark.ml.{Estimator, PipelineModel}
 import org.apache.spark.sql.DataFrame
 import org.apache.spark.sql.types.StructType
 
-class MultiColumnAdapterSpec extends TransformerFuzzingTest {
+import scala.collection.mutable
 
+class MultiColumnAdapterSpec extends EstimatorFuzzingTest {
   val wordDF = session.createDataFrame(Seq(
     (0, "This is a test", "this is one too"),
-    (1, "could be a test", "maybe not"),
-    (2, "foo", "bar")))
+    (1, "could be a test", "bar"),
+    (2, "foo", "bar"),
+    (3, "foo", "maybe not")))
     .toDF("label", "words1", "words2")
-  val inputCols = "words1,words2"
-  val outputCols = "output1,output2"
+  val inputCols = Array[String]("words1","words2")
+  val outputCols = Array[String]("output1", "output2")
 
   test("parallelize transformers") {
     val transformer1 = new Tokenizer()
     val adapter1 =
-      new MultiColumnAdapter().setBaseTransformer(transformer1).setInputCols(inputCols).setOutputCols(outputCols)
-    val tokenizedDF = adapter1.transform(wordDF)
+      new MultiColumnAdapter().setBasePipelineStage(transformer1).setInputCols(inputCols).setOutputCols(outputCols)
+    val tokenizedDF = adapter1.fit(wordDF).transform(wordDF)
     val lines = tokenizedDF.getColAs[Array[String]]("output2")
 
     val trueLines = Array(
       Array("this", "is", "one", "too"),
-      Array("maybe", "not"),
-      Array("bar")
+      Array("bar"),
+      Array("bar"),
+      Array("maybe", "not")
     )
     assert(lines === trueLines)
   }
 
-  override def setParams(fitDataset: DataFrame, transformer: Transformer): Transformer =
-    transformer.asInstanceOf[MultiColumnAdapter]
-      .setBaseTransformer(new Tokenizer())
-      .setInputCols(inputCols)
-      .setOutputCols(outputCols)
+  test("parallelize estimator") {
+    val pipelineStage1 = new StringIndexer()
+    val adapter1 =
+      new MultiColumnAdapter().setBasePipelineStage(pipelineStage1).setInputCols(inputCols).setOutputCols(outputCols)
+    val tokenizedDF = adapter1.fit(wordDF).transform(wordDF)
+
+    val lines1 = tokenizedDF.getColAs[Array[String]]("output1")
+    val trueLines1 = mutable.ArraySeq(1, 2, 0, 0)
+    assert(lines1 === trueLines1)
 
-  override def createDataset: DataFrame = wordDF
+    val lines2 = tokenizedDF.getColAs[Array[String]]("output2")
+    val trueLines2 = mutable.ArraySeq(1, 0, 0, 2)
+    assert(lines2 === trueLines2)
+  }
 
-  override def schemaForDataset: StructType = ???
+  override def createTransformDataset: DataFrame = wordDF
 
```

Comment:
These will become unecessary with the new fuzzer

Reply:
Done, it was great to delete some code